### PR TITLE
refactor: add typings

### PR DESCRIPTION
**Pull request by buskun**
This PR is to add stricter typings to the project.

### CODE REVIEW

**Review by Gusb3ll — APPROVED** [review]

File: .eslintrc.cjs [hunk 1]
```diff
@@ -18,10 +18,7 @@ const config = {
   overrides: [
     {
       files: ['.eslintrc.*', '.prettierrc.*', '*.config.js'],
-      rules: {
-        '@typescript-eslint/no-var-requires': 'off',
-        'no-undef': 'off',
-      },
```
Comment (Leomotors) [inline .eslintrc.cjs:24]: Without this line, there will be something like this. Are there better option to this?

Screenshot 2566-05-17 at 01 23 07

Comment (buskun) [reply .eslintrc.cjs:24]: Hmm, it's not error in my editor tho. We can just disable for only this line. I would not recommend disable a rule for only this purpose.

Comment (Leomotors) [reply .eslintrc.cjs:24]: It only disable the rule for eslintrc and prettierrc file. That should work?

Comment (buskun) [reply .eslintrc.cjs:24]: It should, but I cannot surely confirm as I am not encountering this error. Would you mind trying it out? Just `// eslint-disable-next-line no-undef` should work, if the error is from eslint.

Comment (Leomotors) [reply .eslintrc.cjs:24]: I updated the eslintrc to use { env: { node: true } }
Should follow best practice now :+1:

**Review by Leomotors — APPROVED** [review]

**Pull request merged**